### ISSUE TITLE
Can't Run Files

### ISSUE THREAD

**Issue by swmwy** [andrewbrg/codepad-chrome-app#94]
I think you should add an option to run files so you can test.

**Comment by ghost** [comment]
Yes I can't run files, how could I do that, so I can see what I made?

**Comment by coder0107git** [comment]
What type of file?

**Comment by coder0107git** [comment]
Possibly related/duplicate of #88, #79, #65, #43, #35, #31, #26, #6. 

Also see #92.

**Comment by ghost** [comment]
Yes, I am on chrome OS, and I think my Chromebook is too old to run the Linux environment. I was trying to run a python file.

**Comment by coder0107git** [comment]
How old is too old? I have an old Chromebook manufactured in 2018 and stuck on ChromeOS 103 and it has the Linux container. What version is your Chromebook on?

If you read through all the linked issues you'll find a few resources for how to run python on a Chromebook (online). 

Better yet, use StackBlitz. It has a nicer editor and runs python directly in your browser, works offline too.

**Comment by swmwy** [comment]
Yes, my chromebook cannot enable Linux, so I think it would be smart to make it an actual IDE by allowing it to run all types of files.

**Comment by coder0107git** [comment]
> make it an actual IDE 

This probably won't ever happen. See #92 as for why. 


> allowing it to run all types of files

Chromebooks are secure and locked down. You can't install apps¹, the only app on it is Chrome², and it is physically impossible to access core files or resources³. 

The best/easiest option is to use an online tool like Replit, CodeSandBox, Gitpod, Glitch, StackBlitz, WebVM, Pyodide, PyScript, Sandpack or one of the many other options available. 

WebVM, Sandpack, Pyodide, PyScript, and StackBlitz are by far the most interesting since they run everything directly in your browser. Sandpack, Pyodide, and PyScript are libraries for running Node.js or Python. WebVM runs an entire operating system in the browser which enables running any program in the browser (GUI isn't supported yet and is slow on my Chromebook). 

StackBlitz is the best option since it is closest to an IDE and runs 100% offline. It has out-of-box support for Nodejs and experimental support for Python and WordPress (PHP). There are projects attempting to run Larvel (PHP) among other languages.

If you don't care about running it locally (or offline) you could use Gitpod, CodeSandBox, Replit, Glitch, or one of the many other remote IDEs. 

<br><br>

1 - Apps can't be directly installed on the system. Linux apps are in a container with their own operating system that has only basic access to the main os (files and folders explicitly shared with the app container by the user). Android apps run a literal Android OS directly on the Chromebook with the exact same model as the Linux container. You can "install" PWAs but even though they look like apps they're actually web sites. 

2 - Files, Settings, window manager, networking, etc. are either run directly on Chrome or are heavily tied to Chrome. Google has a project to decouple this called LACROS which will make this less true in a few years when it becomes the default but currently this holds true.

3 - To access core pieces of ChromeOS Dev Mode needs to be enabled which requires removing a read-write screw before being able to access internal system components.

**Comment by hitoyaCute** [comment]
the chromebook i was currently using is stuck on ver 75 so i was also very limited i cant even use crouton and boot any OS, i cant even use a decent website

**Comment by coder0107git** [comment]
@hitoyaCute I'd just say use a raspberry pi. They're like $35 and are pretty decent little computers. The Pi 4 and 5 are more expensive but they are also more powerful than the Pi 3s. They are excellent for developing things like websites, bash scripts, nodejs, rust (if you're willing to wait a long time to compile even the most basic of things), etc. The Pi 4 and the Pi 5 can both run the latest Chromium fairly well too.
